### ISSUE TITLE
Identify a way to fix .epw data in a way BEopt can still load it

### ISSUE THREAD

**Issue by carlosparadis** [erdl/thermal_comfort#19]
Currently proposed methods:

 * Identify a way to convert a fixed .csv into epw (see #17 for epw->csv conversion) OR
 * Identify a way to correct the .epw file while still keeping it loadable by BEopt.

**Comment by carlosparadis** [comment]
@kathrynparadis This may help:

 * https://unmethours.com/question/14109/use-weather-converter-to-convert-csv-to-epw/
 * https://bigladdersoftware.com/projects/elements/

Also for reference:

 * https://beopt.nrel.gov/home

**Comment by kathrynparadis** [comment]
I ended up using the EnergyPlus Weather Statistics and Conversion tool. I found a way to "Override the Default Type" to convert the CSV to EPW without needing the .def file (I just changed the input to the Energy Plus CSV file):

![def file error](https://user-images.githubusercontent.com/30536250/39556670-a6e2d046-4e1d-11e8-8c73-467bac89ac72.png)

![energypluscsv](https://user-images.githubusercontent.com/30536250/39556671-ab90a6b8-4e1d-11e8-9e2d-4b02a4d57f36.png)

**Comment by carlosparadis** [comment]
## Follow-Up

According to Riley, 

<img width="1032" alt="screenshot 2018-05-05 14 14 52" src="https://user-images.githubusercontent.com/17270563/39668639-c6c8ae64-506e-11e8-976d-e481613759fe.png">

A new error is appearing in the converted file. As such, the issue remains open.

**Comment by carlosparadis** [comment]
For the record: This issue was resurrected again a few days ago, as we are trying to reach NREL for a solution. 

Some reference links on EPW:

 * Q&A Forum on .epw creation (https://unmethours.com/question/2393/energy-plus-how-to-create-and-use-epw-and-ddy-files/)

**Comment by carlosparadis** [comment]
@kathrynparadis @eileenpeppard 

## The method to generate the TMY 

The process for **formatting** a csv file a TMY3 file according to that PDF seems easy, however, **creating** is likely impossible for us, or would at least be worth the effort of all we have done up to this point. 

 * A TMY3 file is basically a .csv file with 2 headers from what I understand in the document. 
 * Creating a TMY3 file requires several years of data + a list of special variables to use as input to the algorithm that selects the "most typical month" among all years. Here is a list of them, from the document provided by Riley:

<img width="839" alt="screenshot 2018-07-17 23 22 35" src="https://user-images.githubusercontent.com/17270563/42863988-bb129e7e-8a19-11e8-8188-07498beef4ca.png">

And this doesn't even include the math behind the method of each indicator or the algorithm using them to choose. 

## What I propose

I think this is a lot of time investment for this, and we would be better off trying to exhaust other possibilities first. For instance, I found this tool:

https://beopt.nrel.gov/downloadDView

That serves to visualize **both TMY3 and EPW**. I think @kathrynparadis could use it, or maybe someone on the architecture end who knows BeOPT more than we do, and compare the modified TMY3 to the working .epw to see if we can diagnose why it doesn't work.

@eileenpeppard What do you think?

**Comment by carlosparadis** [comment]
Here are some stuff I noticed you missed by comparing in a notepad the original .csv with the revised .csv (after you save from opening in excel and making the necessary edits:

## 1. Date Format Changes
|Before| After |
|-------|-----|
|<img width="100" alt="screenshot 2018-07-25 00 22 12" src="https://user-images.githubusercontent.com/17270563/43185656-61668a2e-8fa1-11e8-83c5-deae071a0942.png"> | <img width="106" alt="screenshot 2018-07-25 00 22 40" src="https://user-images.githubusercontent.com/17270563/43185663-67b3a66e-8fa1-11e8-8efa-1614d76e5873.png"> |

## 2. Unecessary number of commas (columns) are added

|Before| After |
|-------|-----|
| <img width="1093" alt="screenshot 2018-07-25 00 23 14" src="https://user-images.githubusercontent.com/17270563/43185744-9e88df10-8fa1-11e8-84d1-1bc9fc3bc6b7.png"> |  <img width="1091" alt="screenshot 2018-07-25 00 23 34" src="https://user-images.githubusercontent.com/17270563/43185756-a6072de6-8fa1-11e8-9297-bc1bc2354e7c.png">|

## 3. Decimals have `0.` added as a prefix


**Before**
<img width="954" alt="screenshot 2018-07-25 00 24 00" src="https://user-images.githubusercontent.com/17270563/43185776-b9a40dd8-8fa1-11e8-90b9-b12645c383c7.png">
**After**
<img width="906" alt="screenshot 2018-07-25 00 24 23" src="https://user-images.githubusercontent.com/17270563/43185795-c84c7a96-8fa1-11e8-9995-2e3cd4bd6c10.png">

## Notes

There may be more. I recommend changes to the data are made either in Tableau, R, Python or a text editor like Notepad++ on Windows (dont use the default Notepad).

**Comment by carlosparadis** [comment]
More notes:

Third column from right to left, on the example files provided, the 0.13 value repeated over several rows is replaced by 999.000, the same value, slight different format, from the second column also on right to left.

**Comment by carlosparadis** [comment]
Editing by notepad didnt work even from the original.

Try using this tool to edit the .epw file and see if it works:

https://bigladdersoftware.com/projects/elements/

Dont try to convert to csv ; just edit the epw directly

**Comment by kathrynparadis** [comment]
I used the Elements program, and here is what I found:

1. I am able to edit up to 4 rows (just the Dry Bulb Temperature, and Wet Bulb temperature columns - relative humidity will correct itself), and it will run in BEopt just fine.
2. Any more than 4 rows, and it gives the error again (I tried copying the data from the previous day for the 10 lines of bad data to no avail, and tried duplicating the 1st line of data on each end of the bad data to meet halfway)

The error we get on the new BEopt runs is a bit different than the old (the number on second hers.py line error changed):
![error message with revised epw version 2](https://user-images.githubusercontent.com/30536250/43238860-b6c01a46-902b-11e8-970f-d29d93693b67.png)

**Comment by kathrynparadis** [comment]
Hello all,

I have spoken with Riley and Ben, and it seems that the consensus is that ERI is not required for the BEopt simulations. 

As the error with the edited Barbers point TMY file was related to the ERI, we have decided to just run the simulations without the ERI option marked.

Should we want to use ERI in the future we can look into this error again, however, since this issue is not a priority at the moment we have decided to just run all the simulations without ERI.

Please let us know if you would like us to use the ERI again. 

Thank you,

**Comment by kathrynparadis** [comment]
Just to summarize this:

We ended up editing the tmy file in notepad++. 

At row 4620 (the 1991, 7, 12, 4, 0 row near the middle of the document which represents a July 12th at 4am reading) through row 4629, there were 10 rows (2 columns) of data that had erroneous values of -9.3. The 2 columns in the first 5 rows of bad data were changed to the values of the row directly before the series of bad data (row 4619). The 2 columns in the last 5 rows were replaced with values from the row directly after the bad data (row 4630).

Changing any more than 4 rows while running BEopt with the ERI selected caused it to throw errors and not complete the run, so every file was run without the ERI selected.
